include/seqan3/alignment/pairwise/edit_distance_unbanded.hpp

```diff
@@ -430,20 +430,30 @@ struct pairwise_alignment_edit_distance_unbanded
         {
             get<align_result_key::score>(res) = score();
         }
-        if constexpr (std::tuple_size_v<result_type> >= 3)
+
+        if constexpr (std::tuple_size_v<result_type> == 3)
+        {
+            alignment_coordinate end = end_coordinate();
+            get<align_result_key::end>(res) = end;
+        }
+        else if constexpr (std::tuple_size_v<result_type> == 4)
         {
-            get<align_result_key::end>(res) = end_coordinate();
+            alignment_coordinate end = end_coordinate();
+            trace_matrix_type matrix = trace_matrix();
```

Comment:
is trace_matrix computing or getting the cached trace matrix?

Reply:
computing